cns/service/main.go

```diff
@@ -841,34 +853,60 @@ func main() {
 
 		logger.Printf("Set GlobalPodInfoScheme %v (InitializeFromCNI=%t)", cns.GlobalPodInfoScheme, cnsconfig.InitializeFromCNI)
 
-		err = InitializeCRDState(rootCtx, httpRestService, cnsconfig)
+		err = InitializeCRDState(rootCtx, httpRemoteRestService, cnsconfig)
 		if err != nil {
 			logger.Errorf("Failed to start CRD Controller, err:%v.\n", err)
 			return
 		}
 	}
 
 	// Initialize multi-tenant controller if the CNS is running in MultiTenantCRD mode.
-	// It must be started before we start HTTPRestService.
+	// It must be started before we start HTTPRemoteRestService.
 	if config.ChannelMode == cns.MultiTenantCRD {
-		err = InitializeMultiTenantController(rootCtx, httpRestService, *cnsconfig)
+		err = InitializeMultiTenantController(rootCtx, httpRemoteRestService, *cnsconfig)
 		if err != nil {
 			logger.Errorf("Failed to start multiTenantController, err:%v.\n", err)
 			return
 		}
-
 	}
 
-	logger.Printf("[Azure CNS] Start HTTP listener")
-	if httpRestService != nil {
+	// if user provides cns url by -c option, then only start HTTP remote server using this url
+	logger.Printf("[Azure CNS] Start HTTP Remote server")
+	if httpRemoteRestService != nil {
 		if cnsconfig.EnablePprof {
-			httpRestService.RegisterPProfEndpoints()
+			httpRemoteRestService.RegisterPProfEndpoints()
 		}
 
-		err = httpRestService.Start(&config)
-		if err != nil {
-			logger.Errorf("Failed to start CNS, err:%v.\n", err)
-			return
+		go func() {
+			err = httpRemoteRestService.Start(&config)
```

Comment:
im pretty sure `Start` is not blocking, we can remove the go func wrapping around it here.